python/comparison.cpp

```diff
@@ -83,11 +84,30 @@ template <typename T> void bind_not_equal(py::module &m) {
       docstring_comparison<T>("(x != y)").c_str());
 }
 
+template <typename T> void bind_is_equal(py::module &m) {
+  m.def(
+      "is_equal",
+      [](const typename T::const_view_type &x,
+         const typename T::const_view_type &y) { return x == y; },
+      py::arg("x"), py::arg("y"), py::call_guard<py::gil_scoped_release>(),
+      Docstring()
+          .description("Variable level equals. Returns True if x and y "
+                       "considered equal.")
+          .returns("True if x == y")
+          .rtype<T>()
+          .template param<T>("x", "Input left operand.")
+          .template param<T>("y", "Input right operand.")
+          .c_str());
+}
+
 void init_comparison(py::module &m) {
   bind_less<Variable>(m);
   bind_greater<Variable>(m);
   bind_less_equal<Variable>(m);
   bind_greater_equal<Variable>(m);
   bind_equal<Variable>(m);
   bind_not_equal<Variable>(m);
+  bind_is_equal<Variable>(m);
+  bind_is_equal<Dataset>(m);
+  bind_is_equal<DataArray>(m);
```

Comment:
Wondering why this is not needed for some more cases, just like the old bindings using `bind_comparison`, which included cases of comparisons with views? Is pybind11 capable of implicitly converting *both* arguments to a view?

If so, can we use the same for the bindings of the operators (instead of the additional overloads in `variable.cpp`), so they are defined only for views? Do operators behave differently?